pkg/testsuite/benchmarks_test.go

```diff
@@ -0,0 +1,578 @@
+				"tx1",
+				"Genesis:0",
+				accountIDs...,
+			)
+			// default block issuer issues a block containing the transaction in slot 1.
+			genesisCommitment := iotago.NewEmptyCommitment(ts.API)
+			genesisCommitment.ReferenceManaCost = ts.API.ProtocolParameters().CongestionControlParameters().MinReferenceManaCost
+			block1 := ts.IssueBasicBlockWithOptions("block1", ts.DefaultWallet(), tx1, mock.WithSlotCommitment(genesisCommitment))
+			block = block1.ProtocolBlock()
+			modelBlock := lo.PanicOnErr(model.BlockFromBlock(block))
+			node.Protocol.Events.Engine.Scheduler.BlockScheduled.Hook(func(block *blocks.Block) {
+				blockchan <- block
+			})
+			b.StartTimer()
+			// time from issuance of the block to when it is scheduled
+			node.Protocol.IssueBlock(modelBlock)
+			<-blockchan
+			b.StopTimer()
+			ts.Shutdown()
+		}
+	}
+	// get the ns/op of processing the block
+	nsPerBlock := float64(testing.Benchmark(fn).NsPerOp())
+	fmt.Printf("One input, %d allotments: %f ns/op\n", numAllotments, nsPerBlock)
+	// get the regressors
+	regressors := getBlockWorkScoreRegressors(block)
+	printRegressors(regressors)
+
+	return nsPerBlock, regressors
+}
+
+func printRegressors(regressors []float64) {
+	fmt.Printf("Input: %f\n", regressors[0])
+	fmt.Printf("ContextInput: %f\n", regressors[1])
+	fmt.Printf("Output: %f\n", regressors[2])
+	fmt.Printf("NativeToken: %f\n", regressors[3])
+	fmt.Printf("Staking: %f\n", regressors[4])
+	fmt.Printf("BlockIssuer: %f\n", regressors[5])
+	fmt.Printf("Allotment: %f\n", regressors[6])
+	fmt.Printf("SignatureEd25519: %f\n", regressors[7])
+}
+
+func printCoefficients(coefficients []float64) {
+	minCoeff := math.Abs(coefficients[0])
+	for _, coeff := range coefficients {
+		if math.Abs(coeff) < minCoeff {
+			minCoeff = coeff
+		}
+	}
+	normalisedCoeffs := make([]int, len(coefficients))
+	for i, coeff := range coefficients {
+		normalisedCoeffs[i] = int(coeff / minCoeff)
+	}
+
+	fmt.Println("Calculated coefficients from regression:")
+	fmt.Printf("Block: %d\n", normalisedCoeffs[0])
+	fmt.Printf("Input: %d\n", normalisedCoeffs[1])
+	fmt.Printf("ContextInput: %d\n", normalisedCoeffs[2])
+	fmt.Printf("Output: %d\n", normalisedCoeffs[3])
+	fmt.Printf("NativeToken: %d\n", normalisedCoeffs[4])
+	fmt.Printf("Staking: %d\n", normalisedCoeffs[5])
+	fmt.Printf("BlockIssuer: %d\n", normalisedCoeffs[6])
+	fmt.Printf("Allotment: %d\n", normalisedCoeffs[7])
+	fmt.Printf("SignatureEd25519: %d\n", normalisedCoeffs[8])
+}
+
+func getBlockWorkScoreRegressors(block *iotago.Block) []float64 {
+	regressors := make([]float64, 8)
+
+	basicBlockBody, isBasic := block.Body.(*iotago.BasicBlockBody)
+	if !isBasic {
+		panic("block body is not a basic block body")
+	}
+	signedTx, isSignedTx := basicBlockBody.Payload.(*iotago.SignedTransaction)
+	if !isSignedTx {
+		panic("block payload is not a signed transaction")
+	}
+	// add one to the Input regressor for each input
+	regressors[0] += float64(len(signedTx.Transaction.TransactionEssence.Inputs))
```

Comment:
Maybe we should define an enum for those regressor indexes to avoid mistakes in the future.